service/internal/hamaster/ha_master.go

```diff
@@ -77,6 +77,10 @@ func (h *HAMaster) Mapping(ctx context.Context, obj interface{}) ([]Mapping, err
 		return nil, microerror.Mask(err)
 	}
 
+	if aws.Spec.AvailabilityZones == nil {
+		return nil, microerror.Mask(availabilityZonesNilError)
+	}
+
```

Comment:
Can `aws.Spec` be nil? If yes, what would happen?

Reply:
From what I've seen from the apiextensions I'm assuming this always set. https://github.com/giantswarm/apiextensions/blob/master/pkg/apis/infrastructure/v1alpha2/aws_control_plane_types.go#L53

Reply:
Thanks!